src/intentManager.js

```diff
@@ -219,70 +154,38 @@ define([
         },
         spec: {
             androidIntentAppId: {
                 lstring: 'Intent ID',
                 visibility: 'visible',
                 widget: androidIntentAppId,
-                serialize: function (value, key, mug, data) {
-                    function dropBlank(item) {
-                        item = _.clone(item);
-                        if (item[""] === "") {
-                            delete item[""];
-                        }
-                        return item;
-                    }
-                    var keys = {
-                            "path": _.identity,
-                            "xmlns": function (v) { return v !== DEFAULT_XMLNS ? v : null; },
-                            "extra": dropBlank,
-                            "response": dropBlank,
-                            "unknownAttributes": _.identity,
-                        },
-                        tag = mug.intentTag,
-                        values = {};
-                    if (tag) {
-                        _.each(keys, function (valueOf, attr) {
-                            var val = valueOf(tag.getAttr(attr));
-                            if (!_.isEmpty(val)) {
-                                values[attr] = val;
-                            }
-                        });
-                        if (!_.isEmpty(values)) {
-                            data.intent = values;
-                        }
-                    }
-                },
-                deserialize: function (data, key, mug) {
-                    var tag = makeODKXIntentTag(mug.form, mug.p.nodeID, null);
-                    if (data.intent) {
-                        _.each([
-                            "path",
-                            "xmlns",
-                            "extra",
-                            "response",
-                            "unknownAttributes"
-                        ], function (attr) {
-                            if (!_.isEmpty(data.intent[attr])) {
-                                tag.setAttr(attr, data.intent[attr]);
-                            }
-                        });
-                    }
-                    mug.intentTag = tag;
-                }
             },
             androidIntentExtra: {
                 lstring: 'Extra',
                 visibility: 'visible',
-                widget: androidIntentExtra
+                widget: widgets.baseKeyValue,
+                serialize: serializeAttrs,
             },
             androidIntentResponse: {
                 lstring: 'Response',
                 visibility: 'visible',
-                widget: androidIntentResponse
+                widget: widgets.baseKeyValue,
+                serialize: serializeAttrs,
+            },
+            unknownAttrs: {
+                lstring: 'Unknown',
+                visibility: 'hidden',
+                presence: 'optional',
+                widget: widgets.baseKeyValue,
```

Comment:
Removed. I am keeping the lstring below as I think that lstring is nice for documentation purposes.